src/lib/replication/repl_dev/raft_state_machine.cpp

```diff
@@ -297,14 +297,20 @@ void RaftStateMachine::create_snapshot(nuraft::snapshot& s, nuraft::async_result
 
 int RaftStateMachine::read_logical_snp_obj(nuraft::snapshot& s, void*& user_ctx, ulong obj_id, raft_buf_ptr_t& data_out,
                                            bool& is_last_obj) {
+    if ((obj_id & snp_obj_id_type_mask) == 0) {
```

Comment:
any `obj id < 0x8000000000000000` will result this statement to be true.
I think you want is `obj_id & snp_obj_id_type_mask ==  snp_obj_id_type_mask`

Reply:
For the first bit(type bit), 0 is for HomeStore and 1 is for Application(HomeObject).
So if obj id < 0x8000000000000000, it is expected to be handled by HS.